templates/etc/systemd/system/consul.service.j2

```diff
@@ -0,0 +1,18 @@
+[Unit]
+Description="HashiCorp Consul - A service mesh solution"
+Documentation=https://www.consul.io/
+Requires=network-online.target
+After=network-online.target
+
+[Service]
+Type=notify
+User=consul
+Group=consul
+ExecStart=/bin/bash -c "{{consul_bin}} agent -config-dir={{consul_config_dir}} -datacenter $(curl -s http://169.254.169.254/latest/meta-data/placement/availability-zone/ | rev | cut -c2- | rev)"
```

Comment:
Remember that this is a public role that could be used anywhere (even outside AWS) so the request to metadata endpoint shouldn't be fixed here.

We could use `consul_daemon_options` variable. Make sure to uncomment and set an empty value on `defaults/main.yml`. We could then set our custom value on our private ansible stack